tests/connect/test_with_columns_renamed.py

```diff
@@ -0,0 +1,24 @@
+from __future__ import annotations
+
+
+def test_with_columns_renamed(spark_session):
+    # Test withColumnRenamed
+    df = spark_session.range(5)
+    renamed_df = df.withColumnRenamed("id", "number")
+
+    collected = renamed_df.collect()
+    assert len(collected) == 5
+    assert "number" in renamed_df.columns
+    assert "id" not in renamed_df.columns
+    assert [row["number"] for row in collected] == list(range(5))
+
+    # todo: this fails but is this expected or no?
```

Comment:
Whats the failure?

Reply:
https://github.com/Eventual-Inc/Daft/blob/ba48df20cba5636679c2753295c6f46037c54bdd/tests/connect/test_with_columns_renamed.py#L15-L24